src/main/java/com/arpnetworking/metrics/common/sources/StatsdSource.java

```diff
@@ -0,0 +1,199 @@
+import com.arpnetworking.metrics.mad.parsers.StatsdToRecordParser;
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import jdk.nashorn.internal.runtime.ParserException;
+import net.sf.oval.constraint.NotEmpty;
+import net.sf.oval.constraint.NotNull;
+import net.sf.oval.constraint.Range;
+
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Source that uses Statsd as input.
+ *
+ * @author Ville Koskela (ville dot koskela at inscopemetrics dot com)
+ */
+public final class StatsdSource extends ActorSource {
+
+    @Override
+    protected Props createProps() {
+        return Actor.props(this);
+    }
+
+    /**
+     * Protected constructor.
+     *
+     * @param builder Instance of <code>Builder</code>.
+     */
+    private StatsdSource(final Builder builder) {
+        super(builder);
+        _host = builder._host;
+        _port = builder._port;
+    }
+
+    private final String _host;
+    private final int _port;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StatsdSource.class);
+    private static final Parser<List<Record>, ByteBuffer> PARSER = new StatsdToRecordParser();
+
+    /**
+     * Internal actor to process requests.
+     */
+    /* package private */ static final class Actor extends UntypedActor {
+        /**
+         * Creates a {@link Props} for this actor.
+         *
+         * @param source The {@link StatsdSource} to send notifications through.
+         * @return A new {@link Props}
+         */
+        /* package private */ static Props props(final StatsdSource source) {
+            return Props.create(Actor.class, source);
+        }
+
+        @Override
+        public void onReceive(final Object message) throws Exception {
+            if (Objects.equals(IS_READY, message)) {
+                getSender().tell(_isReady, getSelf());
+            } else if (message instanceof Udp.Bound) {
+                final Udp.Bound updBound = (Udp.Bound) message;
+                LOGGER.debug()
+                        .setMessage("Statsd server binding complete")
+                        .addData("address", updBound.localAddress().getAddress().getHostAddress())
+                        .addData("port", updBound.localAddress().getPort())
+                        .log();
+                _socket = getSender();
+                _isReady = true;
+            } else if (message instanceof Udp.Received) {
+                final Udp.Received updReceived = (Udp.Received) message;
+                LOGGER.debug()
+                        .setMessage("Statsd received datagram")
+                        .addData("bytes", updReceived.data().size())
+                        .log();
+
+                try {
+                    final List<Record> records = PARSER.parse(updReceived.data().toByteBuffer());
+                    records.forEach(_sink::notify);
```

Comment:
This shouldn't be a problem for any parser in my opinion; it could be a problem if the parsing required access to other resources (e.g. disk, services, etc). I will add the comment but can you elaborate on where/why this could be a problem?